[김주영] feat: Chip 주소컴포넌트 구현

kirinnim opened the pull request:
- 기본 박스 컴포넌트
- 주소 선택 컴포넌트
- 주소 chip 컴포넌트
- 기사님용 다중 주소 선택 추가
---
![image](https://github.com/user-attachments/assets/bc833ec1-1d79-4ad3-9adb-9ca1ee6223d8)

kirinnim committed:
feat: Chip 주소컴포넌트 구현
- 기본 박스 컴포넌트
- 주소 선택 컴포넌트
- 주소 chip 컴포넌트

kirinnim committed:
feat: Chip 주소컴포넌트 기사님용 추가

paengdal commented on components/atoms/ChipBubbleTypeBox.tsx:
```diff
@@ -0,0 +1,34 @@
+  text,
+  isSelected = false,
+  onClick,
+  isSelectable = true,
```
- isSelectable props에 대한 설명 정도는 /** */ 주석으로 달아주면 더 좋을 것 같습니다.
- 저같은 경우는 isSelectable이 어떤 속성일까를 한참 고민했는데(삼항 연산자를 보고 알았습니다 ^^;;) isButton이나 canClick등 좀 더 직관적인 표현을 사용하면 좋을 것 같습니다.

paengdal commented on components/atoms/ChipBubbleTypeBox.tsx:
```diff
@@ -0,0 +1,34 @@
+  return (
+    <div
+      className={clsx(
+        'rounded-full px-3 lg:px-5 py-1.5 lg:py-2.5 text-sm lg:text-lg font-medium transition-colors duration-200 border',
```
transition 디테일 좋습니다. 효과가 부드럽네요. :)

paengdal commented on components/atoms/ChipBubbleTypeBox.tsx:
```diff
@@ -0,0 +1,34 @@
+          'cursor-pointer hover:border-[#1B92FF] hover:text-[#1B92FF] hover:bg-[#1B92FF10]',
+        isSelected
+          ? 'border-[#1B92FF] text-[#1B92FF] bg-[#1B92FF10]'
+          : 'border-[#DEDEDE] text-black bg-[#FAFAFA]'
```
- color값은 globals.css에 있는 custom 컬러 변수명을 적용 부탁드립니다. (제가 늦어서 번거롭게 해드리네요. 죄송죄송 ㅠㅠ)
- 다중 선택일 경우 hover의 상태가 선택 상태와 동일하여 선택 해제를 한 건지 사용자가 알기 어렵습니다. hover 효과를 다르게 적용해서 해제 상태를 명확히 알게 하면 더 좋을 것 같습니다.

paengdal commented on components/molecules/RegionDisplay.tsx:
```diff
@@ -0,0 +1,47 @@
+/**
+ * RegionDisplay 컴포넌트
+ *
+ * @description
+ * 선택된 지역을 표시하기 위한 컴포넌트입니다.
+ * 주로 상세 페이지에서 선택된 지역 정보를 보여줄 때 사용합니다.
+ * 단일 지역 또는 여러 지역을 표시할 수 있습니다.
+ *
+ * @example
+ * -단일 지역 표시
+ * const [selectedRegion, setSelectedRegion] = useState<string>('서울');
+ * <RegionDisplay region={selectedRegion} />
+ *
+ * - 다중지역 표시
+ * const [selectedRegions, setSelectedRegions] = useState<string[]>(['서울', '경기']);
+ * <RegionDisplay region={selectedRegions} />
+ *
+ * @param {string|string[]} props.region - 표시할 지역명 또는 지역명 배열
+ */
```
- 구웃~!! 배려가 느껴지는 주석입니다.

paengdal reviewed (approved):
주영님 이제 컴포넌트 아주 깔끔하게 만드시는 것 같습니다.
코드도 아주 깔끔하구요! ㅎㅎ 굿입니다.
그리고 UX에 고민도 하시는게 느껴집니다. 
수고하셨습니다. 홧팅!

eomsung commented on components/molecules/RegionSelector.tsx:
```diff
@@ -0,0 +1,99 @@
+  multipleSelect?: boolean; // 다중 선택 여부
+}
+
+const REGIONS = [
```
배열을 이렇게 설정하신 이유가 있나요??

eomsung commented on components/molecules/RegionSelector.tsx:
```diff
@@ -0,0 +1,99 @@
+  ['충남', '대전', '경북', '대구', '울산'],
+  ['부산', '경남', '전북', '전남', '광주'],
+  ['제주', '세종'],
+];
```
이중 배열로 설정되있는거 같은데 의도된건가요??

kirinnim committed:
perf:간단한 수정 및 지역 타입정의
- region.entitiy.ts 추가
- 정의한 색상표로 적용
- 선택취소 시 효과 추가
- RegionSelector.ts 이중배열에서 배열 슬라이스로 변경
